app/Http/Controllers/Apis/Marketplace/DistributionsApiController.php

```diff
@@ -30,8 +32,98 @@ public function __construct(IDistributionRepository $repository, IResourceServer
         parent::__construct($repository, $resource_server_context);
     }
 
+    #[OA\Get(
+        path: "/api/public/v1/marketplace/distros",
+        description: "Get all marketplace distributions (OpenStack implementations)",
+        summary: 'Get all distributions',
+        operationId: 'getAllDistributions',
+        tags: ['Distributions'],
+        parameters: [
+            new OA\Parameter(
+                name: 'filter[]',
+                in: 'query',
+                required: false,
+                description: 'Filter expressions in the format field<op>value. Available fields: name, company. Operators: =@, ==, @@.',
+                style: 'form',
+                explode: true,
+                schema: new OA\Schema(
+                    type: 'array',
+                    items: new OA\Items(type: 'string', example: 'name@@ubuntu')
+                )
+            ),
+            new OA\Parameter(
+                name: 'order',
+                in: 'query',
+                required: false,
+                description: 'Order by field(s)',
+                schema: new OA\Schema(type: 'string', example: 'name,-id')
+            ),
+            new OA\Parameter(
+                name: 'expand',
+                in: 'query',
+                required: false,
+                description: 'Comma-separated list of related resources to include. Available relations: company, type, capabilities, guests, hypervisors, supported_regions',
+                schema: new OA\Schema(type: 'string', example: 'company,type')
+            ),
+            new OA\Parameter(
+                name: 'relations',
+                in: 'query',
+                required: false,
+                description: 'Relations to load eagerly',
+                schema: new OA\Schema(type: 'string', example: 'company,type')
+            ),
+            new OA\Parameter(
+                name: 'fields',
+                in: 'query',
+                required: false,
+                description: 'Comma-separated list of fields to return',
+                schema: new OA\Schema(type: 'string', example: 'id,name,company.name')
+            ),
+        ],
+        responses: [
+            new OA\Response(
+                response: 200,
+                description: 'Success - Returns paginated list of distributions',
+                content: new OA\JsonContent(
+                    properties: [
+                        'total' => new OA\Property(property: 'total', type: 'integer', example: 20),
```

Comment:
this is already defined on schema as paginate response check https://github.com/OpenStackweb/summit-api/blob/2acb44b2514a7308d774a39cd48214a451d9c6ef/app/Swagger/schemas.php#L58